src/main/java/duke/Task.java

```diff
@@ -0,0 +1,42 @@
+package duke;
+
+abstract class Task {
+    private final int id;
+    private final String name;
+    private boolean isDone = false;
+
+    Task(int id, String name) {
+        this.id = id;
+        this.name = name;
+    }
+
+    Task(int id, String name, Boolean isDone) {
+        this(id, name);
+        this.isDone = isDone;
+    }
+
+    String getName() {
+        return name;
+    }
+
+    void setDone() {
+        isDone = true;
+    }
+
+    void setNotDone() {
+        isDone = false;
+    }
```

Comment:
Not a huge issue but perhaps combining these 2 methods can keep your code more clear and concise?
```suggestion
    void setDone(boolean b) {
        isDone = b;
    }
```